shared/common-adapters/input.native.js

```diff
@@ -0,0 +1,119 @@
+/* @flow */
+
+import React, {Component} from 'react'
+import {TextInput} from 'react-native'
+import {globalColors, globalStyles} from '../styles/style-guide'
+import Platform, {OS} from '../constants/platform'
+import Text from './text.native'
+import Box from './box'
+
+import type {Props} from './input'
+
+type State = {
+  inputFocused: boolean,
+  text: string
+}
+
+// TODO figure out how to make a passwordVisible type
+// We don't want to set this to clear text because that gives
+// permission to the OS to let the text be saved to use for predictive typing.
+
+class Input extends Component<void, Props, State> {
+  state: State;
+  _textInput: any;
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      inputFocused: false,
+      text: this.props.value || ''
+    }
+  }
+
+  _setPasswordVisible (props: Props) {
+    const passwordVisible = props.type === 'passwordVisible'
+    if (this._textInput) {
+      this._textInput.setNativeProps({passwordVisible})
+    }
+  }
+
+  componentWillUpdate (nextProps: Props) {
+    if (nextProps.type !== this.props.type) {
+      this._setPasswordVisible(nextProps)
+    }
+  }
+
+  render () {
+    const IOS = Platform.OS_IOS === OS
+    const inputStyle = Text.textStyle({type: 'BodySemibold'}, {})
+    const password = this.props.type === 'password'
+    const passwordVisible = this.props.type === 'passwordVisible'
+
+    return (
+      <Box style={{...containerStyle, ...this.props.style}}>
+        {this.state.text.length > 0 && <Text type='BodySmall' style={{...floatingLabelStyle}}>{this.props.floatingLabelText}</Text>}
+        <TextInput
+          style={{...inputStyle, ...textInputStyle}}
+          ref={component => { this._textInput = component }}
```

Comment:
If you use `value=` then it is a controlled component meaning it only shows what the prop is that is passed in. It could cause a bit of flickering and janky behavior because it isn't purely native.

Is this something you believe we should add though?